From 963d80d6f9536e20c7bd5be194242d19ba26235b Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Fri, 19 Dec 2025 19:00:24 +0100
Subject: [PATCH] Remove init.vndk-nodef.rc

This partially reverts "Deprecate VNDK-lite support from Legacy GSI".

Paired with https://github.com/LineageOS/android_system_core/commit/0b265e35ddf42638fa807f5349a10c40e3d46446, it kicks permissive devices into bootloader.
Given that GSI+permissive won't ever be a Lineage official scenario, revert this only on our side.
Thanks @Kethen for the insight!

Change-Id: Icb0fecda4b59bb523acbe60b166274c8a6eb689e
---
 target/product/gsi/Android.bp         | 8 --------
 target/product/gsi/init.gsi.rc        | 1 -
 target/product/gsi/init.vndk-nodef.rc | 3 ---
 target/product/gsi_release.mk         | 3 +--
 4 files changed, 1 insertion(+), 14 deletions(-)

diff --git a/target/product/gsi/Android.bp b/target/product/gsi/Android.bp
index 3fa6e76c99..2fee93eb15 100644
--- a/target/product/gsi/Android.bp
+++ b/target/product/gsi/Android.bp
@@ -55,13 +55,6 @@ prebuilt_etc {
     relative_install_path: "init",
 }
 
-prebuilt_etc {
-    name: "init.vndk-nodef.rc",
-    src: "init.vndk-nodef.rc",
-    system_ext_specific: true,
-    relative_install_path: "gsi",
-}
-
 gsi_symlinks = [
     {
         target: "/system/system_ext",
@@ -113,7 +106,6 @@ android_filesystem_defaults {
         ///////////////////////////////////////////
         "gsi_skip_mount.cfg",
         "init.gsi.rc",
-        "init.vndk-nodef.rc",
         // Overlay the GSI specific setting for framework and SystemUI
         "gsi_overlay_framework",
         "gsi_overlay_systemui",
diff --git a/target/product/gsi/init.gsi.rc b/target/product/gsi/init.gsi.rc
index 69c8e467be..94badc8aaf 100644
--- a/target/product/gsi/init.gsi.rc
+++ b/target/product/gsi/init.gsi.rc
@@ -2,4 +2,3 @@
 # Android init script for GSI required initialization
 #
 
-import /system/system_ext/etc/gsi/init.vndk-${ro.vndk.version:-nodef}.rc
diff --git a/target/product/gsi/init.vndk-nodef.rc b/target/product/gsi/init.vndk-nodef.rc
index 1b141a05e7..e69de29bb2 100644
--- a/target/product/gsi/init.vndk-nodef.rc
+++ b/target/product/gsi/init.vndk-nodef.rc
@@ -1,3 +0,0 @@
-on early-init
-    # Reboot if BOARD_VNDK_VERSION is not defined
-    exec - root -- /system/bin/reboot bootloader
diff --git a/target/product/gsi_release.mk b/target/product/gsi_release.mk
index 4855056f8d..dfeef886b6 100644
--- a/target/product/gsi_release.mk
+++ b/target/product/gsi_release.mk
@@ -47,8 +47,7 @@ PRODUCT_USE_DYNAMIC_PARTITION_SIZE := true
 # GSI specific tasks on boot
 PRODUCT_PACKAGES += \
     gsi_skip_mount.cfg \
-    init.gsi.rc \
-    init.vndk-nodef.rc \
+    init.gsi.rc
 
 
 # Overlay the GSI specific setting for framework and SystemUI
-- 
2.25.1

